### ISSUE TITLE
(SDXL?) Lora Training sample preview images blurry / noisy / bad quality

### ISSUE THREAD

**Issue by Lorjuo** [bmaltais/kohya_ss#1456]
Hi,
I got currently the problem that when I am training a lora model using a sdxl checkpoint that I am getting quite blurry, noise or just bad quality preview images.
When I am using the exact same lora file and the same settings for generating the images in automatic1111, then the quality of the images is very good. I did not had this problem beforehand with SD 1.5.
Do you maybe have an idea what could be the reason.
I already tried bf16 and fp16 but both of them resulted in bad preview images. Additionally, I tried quite a lot of different settings without success.

Here one configuration that lead to bad results.
Thanks for your help in advance.

{
  "LoRA_type": "Standard",
  "adaptive_noise_scale": 0,
  "additional_parameters": "",
  "block_alphas": "",
  "block_dims": "",
  "block_lr_zero_threshold": "",
  "bucket_no_upscale": true,
  "bucket_reso_steps": 64,
  "cache_latents": true,
  "cache_latents_to_disk": true,
  "caption_dropout_every_n_epochs": 0.0,
  "caption_dropout_rate": 0,
  "caption_extension": ".txt",
  "clip_skip": 1,
  "color_aug": false,
  "conv_alpha": 1,
  "conv_block_alphas": "",
  "conv_block_dims": "",
  "conv_dim": 1,
  "decompose_both": false,
  "dim_from_weights": false,
  "down_lr_weight": "",
  "enable_bucket": true,
  "epoch": 10,
  "factor": -1,
  "flip_aug": false,
  "full_bf16": false,
  "full_fp16": false,
  "gradient_accumulation_steps": 1,
  "gradient_checkpointing": true,
  "keep_tokens": 1,
  "learning_rate": 0.0003,
  "logging_dir": "",
  "lora_network_weights": "",
  "lr_scheduler": "constant",
  "lr_scheduler_args": "",
  "lr_scheduler_num_cycles": "",
  "lr_scheduler_power": "",
  "lr_warmup": 0,
  "max_bucket_reso": 2048,
  "max_data_loader_n_workers": "",
  "max_resolution": "1024,1024",
  "max_timestep": 1000,
  "max_token_length": "75",
  "max_train_epochs": "",
  "max_train_steps": "",
  "mem_eff_attn": false,
  "mid_lr_weight": "",
  "min_bucket_reso": 256,
  "min_snr_gamma": 5,
  "min_timestep": 0,
  "mixed_precision": "fp16",
  "model_list": "custom",
  "module_dropout": 0,
  "multires_noise_discount": 0,
  "multires_noise_iterations": 0,
  "network_alpha": 16,
  "network_dim": 32,
  "network_dropout": 0,
  "no_token_padding": false,
  "noise_offset": 0,
  "noise_offset_type": "Original",
  "num_cpu_threads_per_process": 2,
  "optimizer": "Adafactor",
  "optimizer_args": "scale_parameter=False relative_step=False warmup_init=False",
  "output_name": "shs_model",
  "persistent_data_loader_workers": false,
  "pretrained_model_name_or_path": ".../sd_xl_base_1.0.safetensors",
  "prior_loss_weight": 1.0,
  "random_crop": false,
  "rank_dropout": 0,
  "reg_data_dir": ".../reg/1024/",
  "resume": "",
  "sample_every_n_epochs": 1,
  "sample_every_n_steps": 0,
  "sample_prompts": "a photo from front of shs woman --n low quality, worst quality, bad anatomy, bad composition, poor, low effort --s 35 --w 768 --h 768 --d 1 --l 7.0",
  "sample_sampler": "dpmsolver++",
  "save_every_n_epochs": 1,
  "save_every_n_steps": 0,
  "save_last_n_steps": 0,
  "save_last_n_steps_state": 0,
  "save_model_as": "safetensors",
  "save_precision": "fp16",
  "save_state": false,
  "scale_v_pred_loss_like_noise_pred": false,
  "scale_weight_norms": 0,
  "sdxl": true,
  "sdxl_cache_text_encoder_outputs": false,
  "sdxl_no_half_vae": true,
  "seed": "58008",
  "shuffle_caption": false,
  "stop_text_encoder_training": 0,
  "text_encoder_lr": 0.0003,
  "train_batch_size": 1,
  "train_data_dir": "...",
  "train_on_input": false,
  "training_comment": "",
  "unet_lr": 0.0003,
  "unit": 1,
  "up_lr_weight": "",
  "use_cp": false,
  "use_wandb": false,
  "v2": false,
  "v_parameterization": false,
  "v_pred_like_loss": 0,
  "vae_batch_size": 0,
  "wandb_api_key": "",
  "weighted_captions": false,
  "xformers": "xformers"
}

**Comment by bparrott78** [comment]
I'm having the same issue.  Preview images look like confetti.

**Comment by sebaxakerhtc** [comment]
You train 1024 res, but have you added 1024 resolution to your samples?
I have a nice quality images during traing because these images are 768x1024, 1024x1024, 832x1280

**Comment by bparrott78** [comment]
Yes. It's not that they're blurry. They're incomprehensible. They look like abstract art.

**Comment by Lorjuo** [comment]
I have now had a look at this phenomenon over multiple trainings / models.
The main observation is that the preview sample "differs" from when I'm using automatic1111 for generating the images with exactly the same settings.
Preview images are always a bit worse.
It seems to be dependent on the training state / model.

- Very good images in automatic1111 -> slightly blurry / noisy images in preview
- slightly overfitted images in automatic1111 -> abstract art, completely random images, ...

Additionally, it also seems not only to make the image blurry / noisy but also changes proportions or replace objects.

I got the feeling that there is maybe some precision loss or rounding in the weights while rendering.

**Comment by Lorjuo** [comment]
I had a look at SD1.5 again and it looks also there like I always had differences between the sample previews and the images generated by automatic1111. Sample previews were always worse, however the notice does not seem to be that significant.

Does somebody ever has achieved comparable quality preview images or even identical ones? If yes, would you mind to send a config to achieve this?

**Comment by lemonliu1992** [comment]
i have the same issue.  it looks Weird when i sample images during training。 but i use lora in webui ， the images generated  looks good。

**Comment by lemonliu1992** [comment]
i set sample_sampler='euler_a' so have better result
